SP23.P03.Web/ClientApp/src/hooks/api/StationDataService.tsx

```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+import { CreateStationDto, TrainStationDto } from "../../types/types";
+import { useEffect, useState } from "react";
+import { User } from "../../types/authentication";
+
+const StationDataService = (user: User | null) => {
```

Comment:
I would advise renaming this to something like `useStations`, since this is a custom React hook. You should be able to do that automatically by just selecting the name in Visual Studio Code, pressing F2, and typing in the new name.

By convention, capitalized names are usually reserved for either React components or Javascript classes, so this looks odd.